compiler/src/ammtoaga/index.ts

```diff
@@ -672,13 +680,29 @@ const loadStatements = (
     vec.push(s);
     line += 1;
   }
+  fnArgs.forEach((arg, i) => {
+    if (globalMem.hasOwnProperty(arg + fnName) && mutatedArgs[arg + fnName]) {
```

Comment:
If the input argument was mutated, re-reference it back to the original argument offset at the end of the closure so it can be picked up properly by the calling opcode, if necessary